src/role_sourcer.js

```diff
@@ -25,6 +25,9 @@ roles.sourcer.settings = {
   },
   amount: {
     300: [1],
+    350: [2],
+    450: [3],
+    550: [4],
```

Comment:
you should then setup maxLayoutAmount[300] to 1 as it shouldn't bug since if you have enouth energy for spawn 2* 'WWWW' then you should already be with capacity > 650 but may cause some unecessary operations